server/api/auth/facebook.js

```diff
@@ -53,25 +68,43 @@ if (clientID) {
   passport.use(new FacebookStrategy(strategyOptions, verifyCallback));
 }
 
+/**
+ * Initiate authentication with Facebook. When the funcion is called, Passport redirects the
+ * user to Facebook to perform authentication.
+ *
+ * @param {*} req Express request object
+ * @param {*} res Express response object
+ * @param {*} next Call the next middleware function in the stack
+ */
 exports.authenticateFacebook = (req, res, next) => {
-  const from = req.query.from ? req.query.from : null;
-  const defaultReturn = req.query.defaultReturn ? req.query.defaultReturn : null;
-  const defaultConfirm = req.query.defaultConfirm ? req.query.defaultConfirm : null;
-
+  const { from, defaultReturn, defaultConfirm } = req.query || {};
   const params = {
-    ...(!!from && { from }),
-    ...(!!defaultReturn && { defaultReturn }),
-    ...(!!defaultConfirm && { defaultConfirm }),
+    ...(from ? { from } : {}),
+    ...(defaultReturn ? { defaultReturn } : {}),
+    ...(defaultConfirm ? { defaultConfirm } : {}),
   };
 
   const paramsAsString = JSON.stringify(params);
 
   passport.authenticate('facebook', { scope: ['email'], state: paramsAsString })(req, res, next);
 };
 
-// Use custom callback for calling loginWithIdp enpoint
-// to log in the user to Sharetribe marketplace with the data from Facebook
+/** This function is called when user returns to this application after authenticating with
+ * Facebook. Passport verifies the recieved tokens and calls a callback function that we've defined
+ * for the authentication strategy and an additional session management function, that we introduce
+ * in this function.
+ *
+ * @param {*} req Express request object
+ * @param {*} res Express response object
+ * @param {*} next Call the next middleware function in the stack
+ */
 exports.authenticateFacebookCallback = (req, res, next) => {
+  // We've already defined the `verifyCallback` function for the Passport Facebook authentication
+  // strategy. That function is normally used to verify the user information obtained from identity
+  // provider, or alternatively create a new use while an internal Passport function is used to
+  // store the user data into session. In our case however, we use the SDK to manage sessions.
+  // Therefore, we provide an additional session management function here, that is called from the
+  // `verifyCallback` fn.
   passport.authenticate('facebook', function(err, user) {
```

Comment:
Did I understand correctly that this anonymous function is the "done" function in verifyCallback?

How about not making it explicit with a function expression?
```js
const doneForVerifyCallback = function(err, user) {
  loginWithIdp(err, user, req, res, clientID, 'facebook');
};
passport.authenticate('facebook', doneForVerifyCallback)(req, res, next);
```

Reply:
If I'm correct, yes. Or it may be that there are some functions in between the `done` from `verifyCallback` and this. There might be Passports own user serialization fns. The Passport.js documentation is not exactly explicit on this. Maybe calling this `sessionFn` etc would convey a correct message.